From 855a1860b2585371e83a3204e3c0d7be76c75138 Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Mon, 16 May 2022 16:16:37 +0300
Subject: [PATCH 10/19] (halium) init: load properties from
 (/system/etc)/prop.halium last

This can be used to override property values set in any other property
files, including "ro." properties.

Change-Id: I8445a6613a471444d5aace295ce049e3b88e19ed
---
 init/property_service.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index fba383c..ca0664c 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -1218,6 +1218,12 @@ void PropertyLoadBootDefaults() {
     load_properties_from_partition("odm", /* support_legacy_path_until */ 28);
     load_properties_from_partition("product", /* support_legacy_path_until */ 30);
 
+    // Halium: load properties from prop.halium last for overrides
+    if (!load_properties_from_file("/system/etc/prop.halium", nullptr, &properties)) {
+        // Try recovery path
+        load_properties_from_file("/prop.halium", nullptr, &properties);
+    }
+
     if (access(kDebugRamdiskProp, R_OK) == 0) {
         LOG(INFO) << "Loading " << kDebugRamdiskProp;
         if (auto res = load_properties_from_file(kDebugRamdiskProp, nullptr, &properties);
-- 
2.45.2

